http-common/src/request.rs

```diff
@@ -186,61 +187,99 @@ where
                 .body(request_body)
                 .expect("cannot fail to create request");
 
-            let err = match tokio::time::timeout(self.timeout, client.request(request)).await {
-                Ok(response) => {
-                    match response {
-                        Ok(response) => break response,
-                        Err(err) => {
-                            if err.is_connect() {
-                                // Network error.
-                                std::io::Error::new(std::io::ErrorKind::NotConnected, err)
-                            } else {
-                                std::io::Error::new(std::io::ErrorKind::Other, err)
+            let (err, backoff_exponential) =
+                match tokio::time::timeout(self.timeout, client.request(request)).await {
+                    Ok(response) => {
+                        match response {
+                            Ok(response) => {
+                                let (
+                                    http::response::Parts {
+                                        status: response_status,
+                                        headers: response_headers,
+                                        ..
+                                    },
+                                    response_body,
+                                ) = response.into_parts();
+
+                                // Make sure to download body inside the timeout
```

Comment:
By "inside the timeout", do you mean gating the entire `to_bytes` within the timeout or just the initial connection as done here?

Reply:
What do you mean by only the initial connection? Before, only the request was inside the timeout, and the   `hyper::body::to_bytes(response_body).await` was outside. I just moved it inside. Now the body download is inside the timeout

Reply:
The body download time is not limited by the timeout with this, though. You would need to wrap the function contents up to the body download with `async { .. }` for that.

Reply:
moved body download into timeout